src/System.Configuration.ConfigurationManager/src/System/Configuration/GenericEnumConverter.cs

```diff
@@ -32,35 +33,54 @@ public override object ConvertFrom(ITypeDescriptorContext ctx, CultureInfo ci, o
             try
             {
                 string value = (string)data;
-                if (string.IsNullOrEmpty(value)) throw new Exception();
+                bool valueIsInvalid = string.IsNullOrEmpty(value);
 
                 // Disallow numeric values for enums.
-                if (!string.IsNullOrEmpty(value) &&
-                    (char.IsDigit(value[0]) ||
-                    (value[0] == '-') ||
-                    (value[0] == '+')))
-                    throw new Exception();
+                if (!valueIsInvalid &&
+                    (char.IsDigit(value[0]) || (value[0] == '-') || (value[0] == '+')))
+                {
+                    valueIsInvalid = true;
+                }
 
-                if (value != value.Trim())
+                if (!valueIsInvalid)
+                {
+                    foreach (char c in value)
+                    {
+                        // throw if the value has whitespace
+                        if (Char.IsWhiteSpace(c))
+                        {
+                            valueIsInvalid = true;
+                            break;
+                        }
+                    }
+                }
+                if (valueIsInvalid)
                 {
-                    // throw if the value has whitespace 
-                    throw new Exception();
+                    throw CreateExceptionForInvalidValue();
                 }
 
                 result = Enum.Parse(_enumType, value);
             }
             catch
             {
-                StringBuilder names = new StringBuilder();
+                throw CreateExceptionForInvalidValue();
```

Comment:
The previous version uses `throw new Exception()` to jump to where the real exception is thrown. Since you're throwing the real exception directly, there's no need to catch it just to build the exception all over again.